### PR TITLE
Add Docker Compose file for local testing

### LINKED ISSUE #186: Support local testing using Docker Compose
## Summary

Hashicorp's [Terraform Provider tutorial](https://developer.hashicorp.com/terraform/tutorials/providers-plugin-framework/providers-plugin-framework-provider) shows you how to use Docker Compose to spin up the application/service you're trying to reach with your Provider. It helps to test your Provider against a real instance of the application to confirm components work as expected.

I'd like to see if we can get this added before getting too far into the Blocks support so we can start testing our changes along the way.

## Acceptance criteria
- [x] A developer can run `terraform` commands against a local instance of Prefect running in Docker to test features of the Prefect Terraform Provider

### PR DESCRIPTION
**Pull request by mitchnielsen**
## Summary

Adds a local Docker Compose file to start an instance of Prefect that we can reach with `terraform` commands.

Closes https://github.com/PrefectHQ/terraform-provider-prefect/issues/186

## Testing

<details>
<summary>Click to expand</summary>

```
$ terraform plan
╷
│ Warning: Provider development overrides are in effect
│
│ The following provider development overrides are set in the CLI configuration:
│  - prefecthq/prefect in /Users/mitch/go/bin
│
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with
│ published releases.
╵

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # prefect_work_pool.example will be created
  + resource "prefect_work_pool" "example" {
      + base_job_template = jsonencode({})
      + created           = (known after apply)
      + default_queue_id  = (known after apply)
      + id                = (known after apply)
      + name              = "my-work-pool"
      + paused            = true
      + type              = "kubernetes"
      + updated           = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply"
now.

~/code/github.com/prefecthq/terraform-provider-prefect/add-testing-in-docker/examples/provider-install-verification on  add-testing-in-docker *?!
$ terraform apply
╷
│ Warning: Provider development overrides are in effect
│
│ The following provider development overrides are set in the CLI configuration:
│  - prefecthq/prefect in /Users/mitch/go/bin
│
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with
│ published releases.
╵

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # prefect_work_pool.example will be created
  + resource "prefect_work_pool" "example" {
      + base_job_template = jsonencode({})
      + created           = (known after apply)
      + default_queue_id  = (known after apply)
      + id                = (known after apply)
      + name              = "my-work-pool"
      + paused            = true
      + type              = "kubernetes"
      + updated           = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

prefect_work_pool.example: Creating...
prefect_work_pool.example: Creation complete after 0s [id=c088c80f-9612-4912-b03b-d5bea4facac7]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

image

</details>

### CODE REVIEW

**Comment by mitchnielsen** [comment]
@parkedwards - looking at the [PREFECT_API_URL in the acceptance tests CI file](https://github.com/PrefectHQ/terraform-provider-prefect/blob/e1bd4cf39a60be016b2f8764d75b422e423fb89e/.github/workflows/acceptance-tests.yaml#L52), I suspect we could replace the value (which I think points to dev or staging) and instead spin up a Prefect server instance using this new Docker Compose setup. That'd give us fresh environments to test against in CI and might help with transient errors related to environment pollution.

Does that sound right/fair? If so I'll open an issue and we can look into it more in a separate PR.

File: compose.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# This file is used to run a local instance of
```
Comment (parkedwards) [inline compose.yml:1]: heh - i did not know they changed this filename convention. TIL

Comment (mitchnielsen) [reply compose.yml:1]: I just recently saw that too. Now that I'm looking at the [docs](https://docs.docker.com/compose/compose-application-model/#the-compose-file), it says `compose.yaml` is preferred but either should work 🤔

File: compose.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# Prefect for testing our Terraform Provider.
+#
+# See ./_about/CONTRIBUTING.md.
+services:
+  prefect:
+    image: prefecthq/prefect:2-latest
```
Comment (parkedwards) [inline compose.yml:7]: i dont think it's out yet, but we may end up updating this to `prefect:3-latest`

**Review by parkedwards — APPROVED** [review]
great idea

**Comment by mitchnielsen** [comment]
> just a caveat - one wrinkle is that our OSS server isn't an exact match/port of our SaaS (which the current acceptance tests are _**mostly**_ oriented towards supporting) - the OSS server has a subset of features and endpoints. so there may be some extra potholes we hit as we try to marry the tests<>this local prefect server instance. but i like where this is going

Great point. I added this note in https://github.com/PrefectHQ/terraform-provider-prefect/issues/192.

File: compose.yml (hunk 2 above)
Comment (mitchnielsen) [reply compose.yml:7]: True, good callout.

**Pull request merged**